app/queries/proxy_rule_query.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class ProxyRuleQuery
+  def initialize(owner_type:, owner_id:, sort: nil, direction: nil, per_page: ThreeScale::Search::Helpers::MAX_PER_PAGE)
+    @direction  = direction
+    @owner_type = owner_type
+    @owner_id   = owner_id
+    @per_page   = per_page.to_i
+    @sort       = sort
+  end
+
+  def search_for(query, scope = ProxyRule.all)
+    scope = scope.order_by(@sort, @direction).includes(:metric)
+    return scope if query.blank?
+    options = {
+      ids_only: true, star: true, per_page: 20, ignore_scopes: true,
```

Comment:
Ouch I dis not check correctly
The per page does not use the variable in. Initialize